### ISSUE TITLE
Doesn't recognise types from php 7.4 typed properties

### ISSUE THREAD

**Issue by hugochinchilla** [jakzal/phpunit-injector#24]
Hello, if using typed properties in php 7.4 you still need to add a redundant annotation for the injector to know the requested type. It would be perfect if it can get the type from the native code.

```php
class ChangeUserPasswordTest extends TestCase
{
    /** @inject */
    private JWTInterface $jwt; // <- should read type from native type declaration
 
    // ....
}
```

**Comment by jakzal** [comment]
Thanks for reporting this!

Indeed PHP 7.4 has not been implemented yet. 

We've been using PHPDocumentor libraries to recognise types. I see few options here:

1. Wait until phpdocumentor recognises PHP 7.4 types.
2. Implement PHP 7.4 specific support based on PHP reflection and tag a new major version.
3. Implement both of the above.

Any thoughts?

**Comment by hugochinchilla** [comment]
I don't understand why adding support to php 7.4 through reflection would require jumping to a new major version but personally I would not have any problems upgrading to a major version and supporting it with reflection seems as a way quicker way to get it done.

I could report this to phpdocumentor and wait for a response but looking at the huge number of open issues I doubt I will get any feedback soon.

**Comment by jakzal** [comment]
s/major/minor/

**Comment by jakzal** [comment]
@hugochinchilla can you have a look at https://github.com/jakzal/injector/pull/16, specifically the test cases, and see if this is what you'd expect?

**Comment by hugochinchilla** [comment]
@jakzal looks perfect, thanks for implementing it so quickly!

**Comment by jakzal** [comment]
Fixed in [v1.4.0](https://github.com/jakzal/phpunit-injector/releases/tag/v1.4.0).

Thanks for sending this feature request!
